src/tracing/builder.rs

```diff
@@ -220,3 +220,124 @@ impl<P: EthereumProvider + Send + Sync + Clone> TracerBuilder<P, Pinned> {
+    async fn test_tracer_builder_block_failure_with_none_block_hash() {
+        // Create a mock Ethereum provider
+        let mut mock_provider = MockEthereumProviderStruct::new();
+        // Expect the chain_id call to return 1
+        mock_provider.expect_chain_id().returning(|| Ok(Some(U64::from(1))));
+        // Expect the block_by_hash call to return an error for an unknown block
+        mock_provider.expect_block_by_hash().returning(|_, _| Ok(None));
+
+        // Create a TracerBuilder with the mock provider
+        let builder = TracerBuilder::new(Arc::new(&mock_provider)).await.unwrap();
+        // Attempt to use the builder with a specific block hash, expecting an error
+        let result = builder.block(BlockId::Hash(B256::repeat_byte(1).into())).await;
+        // Check that the result is an UnknownBlock error
+        assert!(matches!(result, Err(EthApiError::UnknownBlock(_))));
+    }
+
+    #[tokio::test]
+    async fn test_tracer_builder_with_transaction_not_found() {
+        // Create a mock Ethereum provider
+        let mut mock_provider = MockEthereumProviderStruct::new();
+        // Expect the chain_id call to return 1
+        mock_provider.expect_chain_id().returning(|| Ok(Some(U64::from(1))));
+        // Expect the transaction_by_hash call to return Ok(None) for not found transaction
+        mock_provider.expect_transaction_by_hash().returning(|_| Ok(None));
+
+        // Create a TracerBuilder with the mock provider
+        let builder = TracerBuilder::new(Arc::new(&mock_provider)).await.unwrap();
+        // Attempt to use the builder with a specific transaction hash, expecting an error
+        let result = builder.with_transaction_hash(B256::repeat_byte(0)).await;
+        // Check that the result is a TransactionNotFound error
+        assert!(matches!(result, Err(EthApiError::TransactionNotFound(_))));
+    }
+
+    #[tokio::test]
+    async fn test_tracer_builder_with_unknown_block() {
+        // Create a mock Ethereum provider
+        let mut mock_provider = MockEthereumProviderStruct::new();
+        // Expect the chain_id call to return 1
+        mock_provider.expect_chain_id().returning(|| Ok(Some(U64::from(1))));
+        // Expect the transaction_by_hash call to return a transaction with no block number
+        mock_provider
+            .expect_transaction_by_hash()
+            .returning(|_| Ok(Some(Transaction { block_number: None, ..Default::default() })));
+
+        // Create a TracerBuilder with the mock provider
+        let builder = TracerBuilder::new(Arc::new(&mock_provider)).await.unwrap();
+        // Attempt to use the builder with a specific transaction hash, expecting an error
+        let result = builder.with_transaction_hash(B256::repeat_byte(0)).await;
+        // Check that the result is an UnknownBlock error
+        assert!(matches!(result, Err(EthApiError::TransactionNotFound(_))));
+    }
+
+    #[tokio::test]
+    async fn test_tracer_builder_failure() {
```

Comment:
I think this can go, it doesn't really test the builder behaviour, just that the provider throws. Let's just fix this and ok to merge for me!

Reply:
removed, thanks